### PR TITLE
Initial import of new UI (compiled JS code)

### PR DESCRIPTION
**Pull request by tomusdrw**
Signer on `http://127.0.0.1:8180` replaced with new UI.
New Dapps are also accessible, served from `http://localhost:8080` though.

Old Dapps still included, but should be removed in near future.

`parity-dapps-glue` is just a copy of `parity-dapps` in `parity-ui` repository. When we get rid of old dapps it will be renamed to `parity-dapps`.

Related: #2155

### CODE REVIEW

**Comment by jacogr** [comment]
I think it does show us the work still outstanding -

- compilation of the code and making it accessible to the build process (at this point as soon as integrated, it will fall behind unless manually updated)
- actual dapps to be served from a different location when integrated (so they don't go via the ws interface)

I'm wondering if this should be merged or rather just kept open until we can address the above 2 points.

**Comment by tomusdrw** [comment]
1. Yes - that's the tough one. We can utilize existing "compile JS as part of rust build" on CI + "use precompiled js" when building Parity. Where does the pre-compiled JS comes from - thats a separate issue.
2. That one is addressed partially. Dapps are served from `8080`, although the address is hardcoded currently.

My proposal is:
1. `8080 (dapps)` - serving built-in dapps
2. `8081 (dapps+1)` - serving "unsafe" dapps
3. `8180/8546` - WS interface (currently only Signer, in future Pub/Sub + Signer)

**Comment by jacogr** [comment]
The build process is easy to integrate, we already have thing done part-way from a GitLab perspective with running lint & test, next step is just running `npm run build` and pushing the results. 

It is just which location to push this to from CI that is the issue. My suggestion - having a `js-compiled` branch (in the same vein as `gh-pages`) or even a separate `ethcore/js-compiled` repo that we can push to from CI. Whichever makes most sense and is easiest/best from a Rust integration perspective. (Obviously from a JS perspective we want to publish to npm as well in this build script, but that is a separate issue, not affecting the binary builds which is primary concern at this point.)

I would like to hear @gavofyork 's views on this as well, if we can reach agreement on destination the implementation of this "build & make available" will be quick.

I have no issues with your locations suggestions (we have discussed it).

**Comment by gavofyork** [comment]
yeah - happy with @jacogr 's suggestion - pushing precompiled stuff to a separate repo, CI process to build it. will need to work with @General-Beck on this, i guess.

**Comment by General-Beck** [comment]
Yes, we're working on it

**Comment by jacogr** [comment]
We do the have [js-precompiled repo](https://github.com/ethcore/js-precompiled) operational now. It builds and pushes from GitLab to the branch being built, i.e. js branch will push to js, master to master, beta to beta, etc. (Currently since we treat js as the master for js, it is the most up-to-date)

So we can adapt to instead of including the built files in here, pull it from there.

**Comment by jacogr** [comment]
Closes https://github.com/ethcore/parity/issues/2155 when merged along with https://github.com/ethcore/parity/issues/2607 & https://github.com/ethcore/parity/pull/2612.

Tested this branch locally, it new ui loads at http://127.0.0.1:8180, token images show up as expected (content addressing work), new dapps load as expected, sending works.

In short: functionally happy with the PR, it does what it says on the tin.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 7c23bb240e5420c0894a29650c2d7126689209b8
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ethcore/parity

---
This message was auto-generated by https://gitcop.com

File: dapps/js-glue/src/lib.rs.in [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+pub trait WebApp : Default + Send + Sync {
+	fn file(&self, path: &str) -> Option<&File>;
+	fn info(&self) -> Info;
+}
+
+#[cfg(test)]
+mod tests {
+	#[test]
+	fn it_works() {
+	}
```
Comment (gavofyork) [inline dapps/js-glue/src/lib.rs.in]: meant to be empty?

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 7c23bb240e5420c0894a29650c2d7126689209b8
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ethcore/parity

---
This message was auto-generated by https://gitcop.com

**Pull request merged**